src/tests/taskmanager_test.py

```diff
@@ -0,0 +1,84 @@
+"""
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import time
+import sys
+import os
+import unittest
+import random
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+# these are relative imports
+from utils import TaskManager
+from dummytask import DummyTask
+
+
+# run : python -m unittest taskmanager_test.py
+class TestTaskManager(unittest.TestCase):
```

Comment:
This uses delays everywhere, please don't artificially increase the time of tests and use barriers.